meson.build

```diff
@@ -29,13 +29,30 @@ realtime = cc.find_library('rt')
 wayland_client = dependency('wayland-client')
 wayland_protos = dependency('wayland-protocols', version: '>=1.14')
 
-logind = dependency('libsystemd', required: false)
-if logind.found()
-	add_project_arguments('-DHAVE_SYSTEMD=1', language: 'c')
+if get_option('sd-bus-provider') == 'auto'
+	assert(get_option('auto_features').auto(), 'sd-bus-provider must not be set to auto since auto_features != auto')
+	sdbus = dependency('libsystemd',
+		required: false,
+		not_found_message: 'libsystemd not found, trying libelogind',
+	)
+	if not sdbus.found()
+		sdbus = dependency('libelogind',
+			required: false,
+			not_found_message: 'libelogind not found, trying basu',
+		)
+	endif
+	if not sdbus.found()
+		sdbus = dependency('basu',
+			required: false,
+		)
+	endif
+	if not sdbus.found()
+		error('Nor libsystemd, nor libelogind, nor basu was found')
```

Comment:
Should start with "None of" or "Neither". However, would it be better to mark basu required, and use its not_found_message?

Reply:
Oh right, fixed.

> However, would it be better to mark basu required, and use its not_found_message?

I was trying to not make the error message specific to a single potential sd-bus provider (since we try them all).

Reply:
I was thinking you could use this existing error message to replace the default not_found_message and then wouldn't need a final if block. I don't remember if the format that meson prints it out in would look good, though. Either way is fine with me.

Reply:
Looks like this:

```
Run-time dependency basu found: NO (tried pkgconfig and cmake)
Message: Hey there

../meson.build:29:0: ERROR: Dependency "basu" not found, tried pkgconfig and cmake
```

I'm not a fan of the message being in-between two Meson messages which reference "basu".

Here's what happens with the `error` approach:

```
Run-time dependency basu found: NO (tried pkgconfig and cmake)

../meson.build:30:0: ERROR: Problem encountered: Hey there
```

Reply:
Yep, agreed the current way is better. :+1: